src/ngraph/codegen/CMakeLists.txt

```diff
@@ -47,8 +47,14 @@ if ((NGRAPH_GPU_ENABLE OR NGRAPH_CPU_ENABLE) AND NOT NGRAPH_DEX_ONLY)
     list(APPEND HEADER_SEARCH_DEFINES NGRAPH_HEADERS_PATH="${NGRAPH_INCLUDE_PATH}")
 
     if(NGRAPH_DISTRIBUTED_ENABLE)
-        get_target_property(MLSL_INCLUDE_DIR libmlsl INTERFACE_INCLUDE_DIRECTORIES)
-        list(APPEND HEADER_SEARCH_DEFINES MLSL_HEADER_PATH="${MLSL_INCLUDE_DIR}")
+        if (NGRAPH_DISTRIBUTED_MLSL_ENABLE)
+            get_target_property(MLSL_INCLUDE_DIR libmlsl INTERFACE_INCLUDE_DIRECTORIES)
+            list(APPEND HEADER_SEARCH_DEFINES MLSL_HEADER_PATH="${MLSL_INCLUDE_DIR}")
+        else()
+            find_package(MPI REQUIRED)
+            target_compile_definitions(codegen PRIVATE NGRAPH_DISTRIBUTED)
```

Comment:
Why is `NGRAPH_DISTRIBUTED` specific to non-MLSL case? I think we should use separate flags to avoid these issues.

Reply:
removed the use of `NGRAPH_DISTRIBUTED`